src/test/java/com/github/wovnio/wovnjava/ApiTest.java

```diff
@@ -0,0 +1,111 @@
+        Settings settings = makeSettings(new HashMap<String, String>() {{
+            put("projectToken", "token0");
+            put("defaultLang", "en");
+            put("supportedLangs", "en,ja,fr");
+        }});
+        HttpServletRequest request = mockRequestPath("/ja/");
+        String html = translate(request, settings, "<html></html>", 200, "gzip", gzip("{\"body\": \"response html\"}".getBytes()));
+        String expect = "response html";
+        System.out.println(html);
+        assertEquals(expect, html);
+    }
+
+    public void testWithPlainTextResponse() throws ApiException, IOException, ProtocolException {
+        Settings settings = makeSettings(new HashMap<String, String>() {{
+            put("projectToken", "token0");
+            put("defaultLang", "en");
+            put("supportedLangs", "en,ja,fr");
+        }});
+        HttpServletRequest request = mockRequestPath("/ja/");
+        String html = translate(request, settings, "<html></html>", 200, "", "{\"body\": \"response html\"}".getBytes());
+        String expect = "response html";
+        System.out.println(html);
+        assertEquals(expect, html);
+    }
+
+    private String translate(HttpServletRequest request, Settings settings, String html, int code, String encoding, byte[] response) throws ApiException, IOException, ProtocolException {
+        Headers headers = new Headers(request, settings);
+        HttpURLConnection con = mockHttpURLConnection(gzip(html.getBytes()).length, code, encoding, response);
+        Api api = new Api(settings, headers);
+        return api.translate("ja", html, con);
+    }
+
+    private byte[] gzip(byte[] input) throws IOException, ProtocolException {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream(input.length);
+        GZIPOutputStream gz = new GZIPOutputStream(buffer);
+        try {
+            gz.write(input);
+        } finally {
+            gz.close();
+        }
+        return buffer.toByteArray();
+    }
+
+    private HttpURLConnection mockHttpURLConnection( int sendLength, int code, String encoding, byte[] response) throws IOException, ProtocolException {
+        InputStream responseStream = new ByteArrayInputStream(response);
+        HttpURLConnection mock = EasyMock.createMock(HttpURLConnection.class);
+        mock.setDoOutput(true);
+        mock.setRequestProperty(EasyMock.anyString(), EasyMock.anyString());
+        EasyMock.expectLastCall().atLeastOnce();
+        mock.setRequestMethod("POST");
+        EasyMock.expect(mock.getResponseCode()).andReturn(code);
+        EasyMock.expect(mock.getContentEncoding()).andReturn(encoding);
+        EasyMock.expect(mock.getOutputStream()).andReturn(new ByteArrayOutputStream());
+        EasyMock.expect(mock.getInputStream()).andReturn(responseStream);
+        EasyMock.replay(mock);
+        return mock;
+    }
+
+    private HttpServletRequest mockRequestPath(String path) {
```

Comment:
duplicate code with InterceptorTest.java

Reply:
I moved the code to TestUtil.java